hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -149,7 +149,7 @@ int QuectelNcpClient::init(const NcpClientConfig& conf) {
 #if PLATFORM_ID == PLATFORM_B5SOM
     uint32_t hwVersion = HW_VERSION_UNDEFINED;
     auto ret = hal_get_device_hw_version(&hwVersion, nullptr);
-    if (ret == SYSTEM_ERROR_NONE && hwVersion == HAL_VERSION_B5SOM_V003) {
+    if (ret == SYSTEM_ERROR_NONE && hwVersion == HAL_VERSION_B5SOM_V003 && ncpId() == PLATFORM_NCP_QUECTEL_EG91_E) {
```

Comment:
Since we use `&&` here, so just to make sure that ONLY the module that the NCP type is `PLATFORM_NCP_QUECTEL_EG91_E` is going to disable the hardware flow control.

Reply:
Yes, if the NCP type is `PLATFORM_NCP_QUECTEL_EG91_E` and meantime `hwverison` is 0x00 will be blocked hardware flow control to disable.